test/test_plugin_systemd_jruby.rb

```diff
@@ -0,0 +1,27 @@
+require_relative "helper"
+require_relative "helpers/integration"
+
+class TestPluginSystemdJruby < TestIntegration
+
+  THREAD_LOG = TRUFFLE ? "{ 0/16 threads, 16 available, 0 backlog }" :
+    "{ 0/5 threads, 5 available, 0 backlog }"
+
+  def setup
+    skip "Skipped because Systemd support is linux-only" if windows? || osx?
```

Comment:
@mohamedhafez Let's add a new `:linux` symbol to our `skip` logic in `test/helper.rb` and use it here.

Reply:
I know `!windows && !osx != linux` in reality but we can just call it `:linux` for now unless you have a better name 😆  or a better way to detect linux status.

Reply:
I was trying to think through that myself, but in the end figured I'd just stick with `!windows && !osx` for now, since being more exact with that is outside the scope of this PR. Made the changes requested!

Reply:
A bit late to the party, and I can't test any *nix derivatives, but I've wondered about changing the code to something like the below, which gets rid of the squirrelly Windows JRuby code and (since we're Ruby 2.4 +), uses `match?`?
```ruby
HOST_OS = RbConfig::CONFIG['host_os']
IS_WINDOWS = HOST_OS.match?(/mingw|mswin|cygwin/)
IS_OSX     = HOST_OS.include?('darwin')
IS_LINUX   = HOST_OS.include?('linux')
```

Reply:
I'm not familiar enough with what `RbConfig::CONFIG['host_os']` on different Linux systems or Windows platforms to have much of an input on that, but honestly I kind of feel that's beyond the scope of this PR. I just copied the code that was being used for the non-JRuby systemd tests, and made a quick tweak to it at @nateberkopec's request. Could we merge in this PR, and then you guys can optimize that other code in a later PR?

Reply:
> in a later PR

That’s fine.